### PR TITLE
chore/template sync round 2

### PR DESCRIPTION
**Pull request by vmelamed**
# Summary

see below

## Commits

- chore: update vm2.TestUtilities version to 1.4.3 and adjust tag_pattern for semantic versioning
- chore: update changelog for v3.1.0-preview.1 [skip ci]
- chore: update vm2.TestUtilities version to 1.4.3 in package locks

## Checklist

- [x] Commit messages follow [Conventional Commits](https://www.conventionalcommits.org/) format
- [x] Tests pass locally, and cover new code and edge cases. Test coverage is above 80%.
- [x] It is not expected that the performance will degrade by more than 20%
- [x] Documentation is updated if necessary

### CODE REVIEW

File: changelog/cliff.prerelease.toml [hunk 1]
```diff
@@ -26,7 +26,7 @@ footer = ""
 [git]
 conventional_commits = true
 filter_unconventional = true
-tag_pattern = "^v[0-9]+\\.[0-9]+\\.[0-9]+-[a-zA-Z0-9._-]+$"
+tag_pattern = "v[0-9].*"
```
[the remark this answers is not in the document]
Comment (vmelamed) [reply changelog/cliff.prerelease.toml]: k
